app/lib/gpw/client.rb

```diff
@@ -102,5 +107,36 @@ def parse_money_response_nc(response)
         }
       end
     end
+
+    def fetch_shareholders(isin)
+      @client.post('https://www.gpw.pl/ajaxindex.php',
+        {
+          start: 'shareholdersTab',
+          format: 'html',
+          action: 'GPWListaSp',
+          gls_isin: isin,
+        },
+        {
+          'Accept' => '*/*',
+          'DNT' => '1',
+          'User-Agent' => 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:99.0) Gecko/20100101 Firefox/99.0',
+          'Accept-Language' => 'en-US,en;q=0.5',
+        })
+    end
+
+    def parse_shareholders_list(response)
+      doc = Nokogiri::HTML.parse(response.body)
+      table = doc.search('table')[0]
+      table.search('tr').drop(1).map do |tr|
```

Comment:
I think `drop` is mutating the original variable. If there is an alternative way to not mutate it, I would try to use it.